File: goopang-java/src/main/java/com/example/goopangjava/service/UserService.java
package com.example.goopangjava.service;

import com.example.goopangjava.entity.user.User;
import com.example.goopangjava.input.AddUserInput;
import com.example.goopangjava.repository.Database;
import org.apache.coyote.BadRequestException;
import org.springframework.stereotype.Service;

import java.time.OffsetDateTime;
import java.util.List;
import java.util.UUID;

@Service
public class UserService {
    private final CartService cartService;

    public UserService(CartService cartService) {
        this.cartService = cartService;
    }

    public List<User> getUsers() {
        return Database.getInstance().users;
    }

    public User getUser(String userId) throws BadRequestException {
        return Database.getInstance().users.stream()
                .filter(user -> user.getId().equals(userId))
                .findFirst()
                .map(user -> {
                    try {
                        user.setCart(cartService.getUserCart(userId));
                        return user;
                    } catch (BadRequestException e) {
                        throw new RuntimeException(e);
                    }
                })
                .orElseThrow(() -> new BadRequestException("User not found"));
    }

    public User addUser(AddUserInput addUserInput) {
        User user = new User(
                UUID.randomUUID().toString().substring(0, 5),
                addUserInput.getName(),
                addUserInput.getEmail(),
                OffsetDateTime.now()
        );
        Database.getInstance().users.add(user);
        user.setCart(cartService.addUserCart(user));
        return user;
    }
}

File: goopang-kotlin/src/main/kotlin/com/example/goopang/service/UserService.kt
package com.example.goopang.service

import com.example.goopang.entity.user.User
import com.example.goopang.input.AddUserInput
import com.example.goopang.repository.Database
import org.apache.coyote.BadRequestException
import org.springframework.stereotype.Service
import java.time.OffsetDateTime
import java.util.UUID

@Service
class UserService(
    private val cartService: CartService
) {

    fun getUsers(): ArrayList<User> {
        return Database.users
    }

    fun getUser(userId: String): User {
        return Database.users.firstOrNull { it.id == userId }
            ?.also { user -> user.cart = cartService.getUserCart(userId) }
            ?: throw BadRequestException("User not found")
    }

    fun addUser(addUserInput: AddUserInput): User {
        return User(
            id = UUID.randomUUID().toString().substring(0, 5),
            name = addUserInput.name,
            email = addUserInput.email,
            createdAt = OffsetDateTime.now()
        ).also { user -> Database.users.add(user) }
            .also { user -> user.cart = cartService.addUserCart(user) }
    }
}